client/components/Editor/plugins/table.ts

```diff
@@ -24,19 +26,29 @@ function wrapDomSerializer(domSerializer: DOMSerializer) {
  * Synchronize the reactive id attribute of default prosemirror-tables NodeView.
  */
 class PubTableView extends TableView {
-	constructor(node, whatever) {
-		super(node, whatever);
-		this.syncId(node);
+	constructor(node, ...rest) {
+		super(node, ...rest);
+		this.syncDom(node);
 	}
 
 	update(node, decorations) {
 		const shouldUpdate = super.update(node, decorations);
-		this.syncId(node);
+		this.syncDom(node);
 		return shouldUpdate;
 	}
 
-	syncId(node) {
-		((this as any) as { dom: HTMLElement }).dom.setAttribute('id', node.attrs.id);
+	syncDom(node) {
+		const { dom } = (this as any) as { dom: HTMLElement };
+		const label = buildLabel(node);
+		if (label) {
+			const table = dom.querySelector('table');
+			if (table) {
+				const caption = document.createElement('caption');
+				caption.innerHTML = label;
+				table.append(caption);
+			}
+		}
+		dom.setAttribute('id', node.attrs.id);
```

Comment:
Could this code somehow create and append multiple captions to the table? Or is it thrown away/re-created each time table is modified?

Reply:
Oh, that's a good point. I'll factor it into a separate function that only gets called by the constructor.